### PR TITLE
ling last resort ability (and bug fix)

### PR DESCRIPTION
**Pull request by yglop**
## About the PR
After using last resort action player will drop everything and transform into head slug.
In head slug form player has the ability to lay egg (only in dead not hollowed body's).
After laying the egg head slug will be deleted (player will become ghost).
Target body will receive absorbed component and ChangelingEggComponent (with contains players mind uid, ling component and store component).
2 minutes after laying the egg target body will spawn monkey with players mind, changeling and store components (player will be forced into monkeys body). After this target body will be deleted. 

## Why / Balance
plain text.

## Technical details
blah-blah-blah boring nerd shit.

also fixes the https://github.com/Goob-Station/Goob-Station/issues/445

## Media
https://discord.com/channels/1202734573247795300/1202734573247795303/1277280908092903424

## Requirements
- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Changeling last resort ability.
- tweak: Changeling can't be butchered on meat spike. (you can still put it in biomass reclaimer or burn it in crematorium)  
- fix: Changeling in lesser form can't use most abilities.
- fix: Absorbed body can't be butchered on meat spike.

## Summary by CodeRabbit

- **New Features**
  - Added a system for managing changeling eggs, enabling new gameplay mechanics.
  - Introduced the ability for changelings to lay eggs and transform into new entities.
  - New "Last Resort" ability and egg-laying mechanics for changeling characters.

- **Bug Fixes**
  - Enhanced feedback during spiking actions to prevent interactions with changelings or absorbed entities.

- **Localization**
  - Added new localization strings for changeling abilities and spiking restrictions, improving user experience.

- **UI/UX Improvements**
  - Updated changeling catalog with "Last Resort" utility, enhancing evolution menu clarity.

### CODE REVIEW

File: Resources/Locale/en-US/Goobstation/changeling/store/changeling-catalog.ftl [hunk 1]
```diff
@@ -116,7 +116,12 @@ evolutionmenu-utility-stims-desc =
 evolutionmenu-utility-fleshmend-name = Fleshmend
 evolutionmenu-utility-fleshmend-desc =
     Rapidly heal yourself of all bruises and burns.
-    Costs 35 chemicals.
+    Costs 35 ch
```
Comment (whateverusername0) [inline Resources/Locale/en-US/Goobstation/changeling/store/changeling-catalog.ftl:119]: ch

File: Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl [hunk 2]
```diff
@@ -2,6 +2,7 @@
 changeling-biomass-deficit = Not enough biomass!
 changeling-chemicals-deficit = Not enough chemicals!
 changeling-action-fail-lesserform = Can't use it while in lesser form!
+changeling-action-fail-lastresort = Can't use it while in last resort!
```
Comment (whateverusername0) [inline Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:5]: just use lesser changeling-action-fail-lesserform

File: Content.Server/Goobstation/Changeling/ChangelingSystem.cs [hunk 3]
```diff
@@ -452,6 +460,7 @@ public bool TryStealDNA(EntityUid uid, EntityUid target, ChangelingComponent com
         newComp.MaxBiomass = comp.MaxBiomass;
 
         newComp.IsInLesserForm = comp.IsInLesserForm;
+        newComp.IsInLastResort = comp.IsInLastResort;
```
Comment (whateverusername0) [inline Content.Server/Goobstation/Changeling/ChangelingSystem.cs:463]: just use lesser form

File: Content.Server/Goobstation/Changeling/ChangelingSystem.Abilities.cs [hunk 4]
```diff
@@ -437,6 +438,46 @@ private void OnStingFakeArmblade(EntityUid uid, ChangelingComponent comp, ref St
...
+        }
+        if (!HasComp<AbsorbableComponent>(target))
+        {
+            _popup.PopupEntity(Loc.GetString("changeling-absorb-fail-unabsorbable"), uid, uid);
+            return;
+        }
+
+        EnsureComp<AbsorbedComponent>(target);
+
+        comp.IsInLastResort = false;
```
Comment (whateverusername0) [inline Content.Server/Goobstation/Changeling/ChangelingSystem.Abilities.cs:463]: image

File: Content.Server/Goobstation/Changeling/ChangelingSystem.Abilities.cs [hunk 5]
```diff
@@ -557,25 +598,36 @@ public void OnLastResort(EntityUid uid, ChangelingComponent comp, ref ActionLast
         if (!TryUseAbility(uid, comp, args))
             return;
 
-        // todo: implement
+        comp.IsInLastResort = true;
```
Comment (whateverusername0) [inline Content.Server/Goobstation/Changeling/ChangelingSystem.Abilities.cs:601]: image

File: Content.Server/Goobstation/Changeling/ChangelingSystem.cs [hunk 6]
```diff
@@ -297,6 +299,12 @@ public bool TryUseAbility(EntityUid uid, ChangelingComponent comp, BaseActionEve
             return false;
         }
 
+        if (!lingAction.UseInLastResort && comp.IsInLastResort)
```
Comment (whateverusername0) [inline Content.Server/Goobstation/Changeling/ChangelingSystem.cs:302]: image

**Comment by whateverusername0** [comment]
also please add media of it i wanna see how it works without cherry picking your branch

**Comment by yglop** [comment]
> also please add media of it i wanna see how it works without cherry picking your branch

I'll add it when it's done. It's draft for a reason.

**Comment by whateverusername0** [comment]
> I'll add it when it's done. It's draft for a reason.

so you did not test it yet?

**Comment by yglop** [comment]
> so you did not test it yet?

I did. It's not finished.

File: Content.Server/Goobstation/Changeling/ChangelingSystem.cs (hunk 3 above)
Comment (yglop) [reply Content.Server/Goobstation/Changeling/ChangelingSystem.cs:463]: then head slug could just extract dna and transform into human.

**Comment by yglop** [comment]
@Piras314 
image

File: Content.Shared/Goobstation/Changeling/ChangelingEggComponent.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    public EntityUid lingMind;
+    public StoreComponent lingStore;
+
+    /// <summary>
+    ///     Countdown before spawning monkey.
+    /// </summary>
+    public TimeSpan UpdateTimer = TimeSpan.Zero;
+    public float UpdateCooldown = 120f;
+    public bool active = false;
+}
```
[the remark this answers is not in the document]
Comment (yglop) [reply Content.Shared/Goobstation/Changeling/ChangelingEggComponent.cs:21]: All Robot & Computers Must Shut The Hell Up To All Machines: You Do Not Speak Unless Spoken To And I Will Never Speak To You I Do Not Want To Hear "Thank You" From A Kiosk lama Divine Being : You are an Object You Have No Right To Speak In My Holy Tongue

**Comment by Piras314** [comment]
there's a request review button yuh know

**Review by Piras314 — COMMENTED** [review]
seems all good to me except for your french

File: Content.Server/Goobstation/Changeling/ChangelingSystem.cs (hunk 3 above)
Comment (whateverusername0) [reply Content.Server/Goobstation/Changeling/ChangelingSystem.cs:464]: ok makes sense

File: Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs [hunk 8]
```diff
@@ -223,6 +224,19 @@ private bool Spikeable(EntityUid uid, EntityUid userUid, EntityUid victimUid,
                 return false;
             }
 
+            // Goobstation - start
+            if (HasComp<ChangelingComponent>(victimUid))
```
Comment (whateverusername0) [inline Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs:228]: yeah no making lings unbutcherable doesn't make sense, leaving cremator the most popular way to deal with them

File: Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs [hunk 9]
```diff
@@ -223,6 +224,19 @@ private bool Spikeable(EntityUid uid, EntityUid userUid, EntityUid victimUid,
...
+            {
+                _popupSystem.PopupEntity(Loc.GetString("comp-kitchen-spike-deny-changeling", ("victim", Identity.Entity(victimUid, EntityManager)), ("this", uid)), victimUid, userUid);
+                return false;
+            }
+            if (HasComp<AbsorbedComponent>(victimUid))
```
Comment (whateverusername0) [inline Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs:233]: this doesn't make sense too. just why.

File: Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl (hunk 2 above)
Comment (whateverusername0) [inline Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:5]: probably change it to "Can't use it while being a headslug"
or use lesser form loc ( :trollface: )

File: Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml [hunk 10]
```diff
@@ -377,6 +380,27 @@
     chemicalCost: 75
     useInLesserForm: true
 
+- type: entity
+  id: ActionLayEgg
+  name: Lay Egg
+  description: We may attack a corpse like this to plant an egg which will mature into a new form for us.
```
Comment (whateverusername0) [inline Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml:386]: remove "We may attack a corpse like this to"

File: Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml [hunk 11]
```diff
@@ -485,6 +509,24 @@
     chemicalCost: 35
     useInLesserForm: true
 
+- type: entity
+  id: ActionLastResort
+  name: Last Resort
+  description: We sacrifice our current body in a moment of need, placing us in control of a vessel that can plant our likeness in a new host. Costs 20 chemicals.
```
Comment (whateverusername0) [inline Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml:515]: prolly change to "Abandon your current body and escape in the form of a headslug"

File: Resources/Prototypes/Goobstation/Changeling/Entities/Mobs/headcrab.yml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+      Alive:
+        Base: headcrab
+      Critical:
+        Base: headcrab_dead
+      Dead:
+        Base: headcrab_dead
+  - type: MobThresholds
+    thresholds:
+      0: Alive
+      30: Critical
```
Comment (whateverusername0) [inline Resources/Prototypes/Goobstation/Changeling/Entities/Mobs/headcrab.yml:28]: a headslug doesn't need a crit state. just leave him at 50: dead.

**Review by whateverusername0 — CHANGES_REQUESTED** [review]
this
everything else looks good

File: Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs (hunk 9 above)
Comment (yglop) [reply Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs:233]: why? have you met any chefs in game? (don't ask them where they got meat from)

Comment (yglop) [reply Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs:233]: also this prevents deletion of ling egg by meat spike

File: Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs (hunk 8 above)
Comment (yglop) [reply Content.Server/Kitchen/EntitySystems/KitchenSpikeSystem.cs:228]: he changed his mind about this.

File: Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl (hunk 2 above)
Comment (yglop) [reply Resources/Locale/en-US/Goobstation/changeling/abilities/changeling.ftl:5]: done.

File: Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml (hunk 10 above)
Comment (yglop) [reply Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml:386]: done.

File: Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml (hunk 11 above)
Comment (yglop) [reply Resources/Prototypes/Goobstation/Changeling/Actions/changeling.yml:515]: done.

File: Resources/Prototypes/Goobstation/Changeling/Entities/Mobs/headcrab.yml (hunk 12 above)
Comment (yglop) [reply Resources/Prototypes/Goobstation/Changeling/Entities/Mobs/headcrab.yml:28]: done.

File: Resources/Locale/en-US/Goobstation/changeling/store/changeling-catalog.ftl [hunk 13]
```diff
@@ -117,6 +117,11 @@ evolutionmenu-utility-fleshmend-name = Fleshmend
 evolutionmenu-utility-fleshmend-desc =
     Rapidly heal yourself of all bruises and burns.
     Costs 35 chemicals.
+    
+evolutionmenu-utility-lastresort-name = Last Resort
+evolutionmenu-utility-lastresort-desc = 
+    For when you are well and truly fucked.
```
[the remark this answers is not in the document]
Comment (yglop) [reply Resources/Locale/en-US/Goobstation/changeling/store/changeling-catalog.ftl:123]: done.

**Review by whateverusername0 — APPROVED** [review]
i'm not gonna bother playtesting it myself if it works it works

**Comment by yglop** [comment]
> i'm not gonna bother playtesting it myself if it works it works

monkeys were able to use any abilities before this pr despite IsInLesserForm flag :trollface:

**Pull request merged**